scalyr_agent/configuration.py

```diff
@@ -50,7 +51,75 @@ class Configuration(object):
     This also handles reporting status information about the configuration state, including what time it was
     read and what error (if any) was raised.
     """
-    def __init__(self, file_path, default_paths):
+
+    # Authoritative list of params that can be overridden by environment variables.
+    ENVIRONMENT_AWARE_CONFIG_PARAMS = {
+        # Special-case global params
+        'api_key': None,
+        'scalyr_server': 'SCALYR_SERVER',
```

Comment:
Except for `scalyr_server`, all other params don't need custom names ( can default to prepending `SCALYR_` )